src/Symfony/Component/EventDispatcher/EventDispatcher.php

```diff
@@ -233,4 +247,28 @@ private function sortListeners($eventName)
             }
         }
     }
+
+    /**
+     * Optimizes the internal list of listeners for the given event by priority.
+     */
+    private function optimizeListeners(string $eventName): array
+    {
+        krsort($this->listeners[$eventName]);
+        $optimized = &$this->optimized[$eventName];
+        $optimized = array();
+
+        foreach ($this->listeners[$eventName] as &$listeners) {
+            foreach ($listeners as &$listener) {
+                $closure = &$optimized[];
+                $closure = $listener instanceof \Closure ? $listener : static function (...$args) use (&$listener, &$closure) {
```

Comment:
I don't think this intermediary closure is needed

Reply:
this closure allows creating the listener services lazily: right now, all listener services are created even if one of them stops propagation

Reply:
Yes but listeners rarely stop propagation. I think we should avoid doing these "cache misses" and assume most listeners are going to be executed.

Reply:
why? to save the closure? it doesn't have any impact. Laziness is much more desirable to me here.

Reply:
oh, actually we can move the "if" outside of the closure and save creating a closure when no lazy listeners are involved. updated.